docs/architecture/concentrated-liquidity-adr.md

```diff
@@ -238,19 +261,292 @@ See the next `"Swap Router Module"` section of this document for more details.
 With the new `concentrated-liquidity` module, we now have a new entrypoint for swaps that is
 the same with the existing `gamm` module.
 
-To avoid fragmenting swap entrypoints and duplicating boilerplate logic, we would like to define
-a new `swap-router` module. For now, its only purpose is to receive swap messages and propagate them
-either to the `gamm` or `concentrated-liquidity` modules.
+To avoid fragmenting swap and pool creation entrypoints and duplicating their boilerplate logic,
+we would like to define a new `swaprouter` module. Its purpose is twofold:
+1. Handle pool creation messages
+   * Assign ids to pools
+   * Store the mapping from pool id to one of the swap modules (`gamm` or `concentrated-liquidity`)
```

Comment:
Nit: consider changing this to pool module, it confused me at first with the thought that the "swap module" is referring to the swap router module
```suggestion
   * Store the mapping from pool id to one of the pool modules (`gamm` or `concentrated-liquidity`)
```

Reply:
Let's revisit this separately please. I agree with you on this but was planning to make a separate issue to consistently change the naming everywhere, including the docs

Reply:
Created an issue: https://github.com/osmosis-labs/osmosis/issues/3241